src/Servers/IIS/IIS/src/Core/IISHttpContext.Features.cs

```diff
@@ -75,6 +77,7 @@ private void Initialize()
         _currentIServerVariablesFeature = this;
         _currentIHttpMaxRequestBodySizeFeature = this;
         _currentITlsConnectionFeature = this;
+        _currentITlsHandshakeFeature = this;
```

Comment:
Only set this if https. Otherwise null. The same goes for ITlsConnectionFeature. See how IHttpResponseTrailersFeature is also conditional.

Reply:
updated, thanks.